src/ILCompiler.Compiler/src/Compiler/DebugInformationProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.IL;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Provides debug information by delegating to the <see cref="MethodIL"/>.
+    /// </summary>
+    public class DebugInformationProvider
+    {
+        public virtual bool IsNull => false;
```

Comment:
This doesn't really make sense here. Its more of a per methodIL property ... if you just wanted IsNull like this... just use  ... is NullDebugInformationProvider via the typesystem instead of adding an api.